packages/core/src/cache/classes/ImageVolume.ts

```diff
@@ -16,7 +29,7 @@ export class ImageVolume implements IImageVolume {
   /** volume origin, Note this is an opinionated origin for the volume */
   origin: Point3;
   /** volume scalar data  */
-  scalarData: Float32Array | Uint8Array;
+  scalarData: VolumeScalarData | Array<VolumeScalarData>;
```

Comment:
add a comment on top saying 3D or 4D

Reply:
should'nt scalarData also be private? since they should be accessed via `.getScalarData`?

Reply:
Yes and I totally agree but making that change will break any code from other devs that are already working on cs3d. May I remove it from the interface?

Reply:
Updated the '.scalarData` from `public` to `protected` because children classes access it and fixed some other codes that were accessing it directly.